### PR TITLE
DEVEXP-62: Use ExecuteScriptMarkLogic with consistent behavior

### PR DESCRIPTION
**Pull request by AndreiPanti**
Thank you for submitting a contribution to marklogic-community/marklogic-nifi-incubator.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a github ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### CODE REVIEW

**Comment by AndreiPanti** [comment]
I encountered some issues regarding the second point. Once I ran the test (even without any modification of my modifications I received some connection errors. Other tests work fine, others are producing the same issue.
I attached some screenshots with the errors.


ErrorPart1
ErrorPart2

File: nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/ExecuteScriptMarkLogic.java [hunk 1]
```diff
@@ -241,9 +258,22 @@ public final void onTrigger(final ProcessContext context, final ProcessSessionFa
                 }
             }
 
+            String scriptBody = determineScriptBody(context, originalFF);
```
Comment (rjrudin) [inline nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/ExecuteScriptMarkLogic.java:261]: There's already logic in `buildCall` to determine if the user is using a script or a module. Should just modify that to add the appropriate attribute to the original FlowFile.

File: nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/ExecuteScriptMarkLogic.java [hunk 2]
```diff
@@ -161,7 +178,7 @@ public ValidationResult validate(String subject, String input, ValidationContext
         .description("Receives the original FlowFile if the call to MarkLogic fails for any reason").build();
 
 
-    private static Charset UTF8 = StandardCharsets.UTF_8;
+    private static Charset UTF8 = Charset.forName("UTF-8");
```
Comment (rjrudin) [inline nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/ExecuteScriptMarkLogic.java:181]: This is undoing some changes from #170 - please make sure the PR only has changes needed for DEVEXP-62.

**Review by rjrudin — CHANGES_REQUESTED** [review]
Still need a test to verify requirement 2 - the addition of the new attribute. Also, requirement 1 - not requiring a FlowFile to be run - hasn't been addressed.

**Pull request closed without merging**